tests/utilities/test_common.py

```diff
@@ -0,0 +1,97 @@
+import pandas as pd
+import numpy as np
+from hypothesis import given, settings, strategies as st
+from hypothesis.extra.pandas import column, data_frames, range_indexes
+from mawpy.constants import STAY_LAT, STAY_LONG, STAY
+from mawpy.utilities.common import _mean_ignore_minus_ones, _merge_stays, get_combined_stay, get_stay_groups
+
+# Testing _mean_ignore_minus_one()
+series_strategy = st.lists(st.one_of(st.integers(min_value=-1, max_value=100), st.just(-1)),
+                            min_size=1,
+                            max_size=100).map(pd.Series)
+@given(series=series_strategy)
+def test_mean_ignore_minus_ones(series):
+    result = _mean_ignore_minus_ones(series)
+    if all(series == -1):
+        assert result == -1
+    else:
+        assert result == series[series != -1].mean()
+
+# Testing _merge_stays()
+df_columns = {
+    STAY: {"elements": st.integers(min_value=0, max_value=10)},
+    STAY_LAT: {"elements": st.floats(min_value=-90, max_value=90)},
+    STAY_LONG: {"elements": st.floats(min_value=-180, max_value=180)}
+}
+df_strategy = data_frames(
+    index=range_indexes(min_size=5, max_size=10),
+    columns=[column(key, **value) for key, value in df_columns.items()]
+)
+@given(df_by_user=df_strategy, group_avgs=df_strategy, stay_to_update=st.integers(min_value=0, max_value=10),
+       updated_stay=st.integers(min_value=0, max_value=10), group_avgs_index_to_update=st.integers(min_value=0, max_value=10))
+@settings(max_examples=100, deadline=None)
+def test_merge_stays(df_by_user, group_avgs, stay_to_update, updated_stay, group_avgs_index_to_update):
+
```

Comment:
For other required params - you can use something like this:

`stay_to_update = st.sampled_from(unique_stays).map(int)
    updated_stay = st.sampled_from(unique_stays).map(int)
    group_avgs = df_by_user.groupby(STAY)[[STAY_LAT, STAY_LONG]].mean().reset_index()
    group_avgs_index_to_update = group_avgs.loc[group_avgs[STAY] == stay_to_update].index

    df_by_user_expected = df_by_user.copy()
    group_avgs_expected = group_avgs.copy()`